devito/types/dimension.py

```diff
@@ -536,6 +535,77 @@ def _arg_check(self, *args, **kwargs):
 # The Dimensions below are exposed in the user API. They can only be created by
 # the user
 
+class Thickness(DataSymbol):
+    """A DataSymbol to represent a thickness of a SubDimension"""
+
+    __rkwargs__ = DataSymbol.__rkwargs__ + ('root', 'side', 'local', 'value')
+
+    def __new__(cls, *args, root=None, side=None, local=False, **kwargs):
+        newobj = super().__new__(cls, *args, **kwargs)
+        newobj._root = root
+        newobj._side = side
+        newobj._local = local
+
+        return newobj
+
+    def __init_finalize__(self, *args, **kwargs):
+        self._value = kwargs.pop('value', None)
+
+        kwargs.setdefault('is_const', True)
+        super().__init_finalize__(*args, **kwargs)
+
+    @property
+    def root(self):
+        return self._root
+
+    @property
+    def side(self):
+        return self._side
+
+    @property
+    def local(self):
+        return self._local
+
+    @property
+    def value(self):
+        return self._value
+
+    def _arg_check(self, *args, **kwargs):
+        # TODO: Should check that the thickness isn't larger than the parent
+        # domain
+        pass
+
+    def _arg_values(self, grid=None, **kwargs):
+        # Allow override of thickness values to disable BCs
+        # However, arguments from the user are considered global
+        # So overriding the thickness to a nonzero value should not cause
+        # boundaries to exist between ranks where they did not before
+        r_tkn = kwargs.get(self.name, self.value)
+        if grid is not None and grid.is_distributed(self.root):
+            # Get local thickness
+            if self.local:
+                # Dimension is of type `left`/`right` - compute the offset
+                # and then add 1 to get the appropriate thickness
+                if self.value is not None:
+                    tkn = grid.distributor.glb_to_loc(self.root, r_tkn-1, self.side)
+                    tkn = tkn+1 if tkn is not None else 0
+                else:
+                    tkn = 0
+            else:
+                # Dimension is of type `middle`
+                tkn = grid.distributor.glb_to_loc(self.root, r_tkn, self.side) or 0
+        else:
+            tkn = r_tkn or 0
+
+        return {self.name: tkn}
+
+    def _arg_finalize(self, *args, **kwargs):
+        return {}
+
+    def _arg_apply(self, *args, **kwargs):
```

Comment:
drop this, irrelevant

Reply:
A no-op `_arg_check`, `_arg_finalize`, and `_arg_apply` are required for `_prepare_arguments`

Reply:
this should be fixed at the superclass site then, somehow, but fine..